shared/domains/chelonia/internals.js

```diff
@@ -253,7 +253,22 @@ export default (sbp('sbp/selectors/register', {
       if (latest !== recent) {
         console.debug(`[chelonia] Synchronizing Contract ${contractID}: our recent was ${recent || 'undefined'} but the latest is ${latest}`)
         // TODO: fetch events from localStorage instead of server if we have them
-        const events = await sbp('chelonia/out/eventsSince', contractID, recent || contractID)
+        const events = await sbp('chelonia/out/eventsAfter', contractID, recent || contractID)
+        // checks if the list of events consist of latest event
+        // TODO: if we use findLastIndex, it will be more clean
+        //       but it needs upgrade Cypress version to 9.7.0 which is of bad performance
+        //       https://docs.cypress.io/guides/references/changelog#9-7-0
+        //       https://github.com/cypress-io/cypress/issues/22868
+        let isLatestExistance = false
+        for (let i = events.length - 1; i >= 0; i--) {
+          if (GIMessage.deserialize(events[i]).hash() === latest) {
+            isLatestExistance = true
+            break
+          }
+        }
+        if (!isLatestExistance) {
+          throw new ChelErrorUnexpected()
+        }
```

Comment:
@taoeffect, not sure if we should use `findLastIndex` here, and make it much smart.
Regarding the browser compatibility of `findLastIndex`, we should upgrade Cypress version to 9.7.0 or use the Chrome (not Electron) as the default browser in the current version 7.7.0.

I would like to hear your opinion on this.

Reply:
### References
- `findLastIndex` Browser Compatibility
![image](https://github.com/okTurtles/group-income/assets/57976479/5ce7d008-3821-4bfb-babd-3b53296ac3b3)
- Cypress v7.7.0 browser versions: 
Electron: Chromium **94.0.4606.81**, Chrome: **113**, Firefox: **113**
![image](https://github.com/okTurtles/group-income/assets/57976479/e62fd60a-626e-4c7b-9e77-09a4bb9fdc68)
- Cypress v9.7.0 Change Log and Browser versions: 
https://docs.cypress.io/guides/references/changelog#9-7-0
Electron: Chromium **100.0.4896.75**
- Cypress Bad Performance issue
https://github.com/cypress-io/cypress/issues/22868